pycqed/measurement/waveform_control/pulsar.py

```diff
@@ -532,6 +544,15 @@ def _stop_AWG(self, AWG):
         else:
             raise ValueError('Unsupported AWG type: {}'.format(type(obj)))
 
+    def _is_AWG_running(self, AWG):
+        obj = self.AWG_obj(AWG=AWG)
+        if isinstance(obj, Tektronix_AWG5014):
+            return obj.get_state() != 'Idle'
+        elif isinstance(obj, UHFQC):
+            raise NotImplementedError()
+        else:
+            raise ValueError('Unsupported AWG type: {}'.format(type(obj)))
```

Comment:
I'd say that Unsupported AWG type should be a NotImplementedError (with exactly that message). It may also be beneficial to add the name of the object so that people know which of their AWG's is of the offending type. 

Also worth checking if classes that inherit from this (say maybe some other AWG5014 sub type) would raise an error here. I'm mostly thinking of this in the context of dummy instruments.

Reply:
The `else` case is meant for the case when some object is (accidentally) specified as the AWG for the Pulsar that is not meant to be supported. So I consider `ValueError` more appropriate than `NotImplementedError`.